doc/specs/#476 - Pinned packages.md

```diff
@@ -0,0 +1,186 @@
+
+```
+PS C:\Users\billg> winget install --pin Microsoft.Bob
+Found Microsoft Bob [Microsoft.Bob] Version 2.32.0
+This application is licensed to you by its owner.
+Microsoft is not responsible for, nor does it grant any licenses to, third-party packages.
+Downloading https://aka.ms/importantsystemdependencies/bob.exe
+██████████████████████████████ 2.08 MB / 2.08 MB
+Successfully verified installer hash
+Starting package install...
+Successfully installed
+Microsoft.Bob has been pinned at version 2.32.0.
+```
+
+Then, later, that user tries to install a package that is dependent on a newer version of Microsoft Bob. The Windows Package Manager should not upgrade Microsoft Bob automatically, but instead should notify the user and give them information on how to continue:
+
+```
+PS C:\Users\billg> winget install --pin Microsoft.RoverTheDog
+Found Rover the Dog [Microsoft.RoverTheDog] Version 5.1
+This application is licensed to you by its owner.
+Microsoft is not responsible for, nor does it grant any licenses to, third-party packages.
+This package requires the following dependencies:
+  - Packages
+      Microsoft.Bob, version 2.39.0.
+
+Microsoft Bob is currently pinned at version 2.32.0, but Rover the Dog 5.1 requires at least version 2.39.0. 
+To upgrade Microsoft Bob and other pinned dependencies, add the argument --include-pinned to your previous command.
+```
+
+## Capabilities
+
+### Accessibility
+
+Accessibility should not be impacted by this change. There will be a few more tables printed to the terminal in certain cases, but they should use the current table implementation used by `winget upgrade` and `winget list`.
+
+### Security
+
+Security of the Windows Package Manager should not be impacted by this change. However, security of user's software may be, as if they pin a insecure version of a package they currently will not be notified of important security updates within the Windows Package Manager.
+
+### Reliability
+
+The change will improve reliability, as users will be able to have fine grained control of the Windows Package Manager's upgrade functionality to ensure their workflow is not disrupted. 
+
+### Compatibility
+
+There should not be any breaking changes to the code. Although there could be a mild breaking change to the behavior of `upgrade --all` (not all packages are upgraded anymore since pinned ones are skipped), this is purely opt-in from the user's perspective at this time (if they do not pin software, there should not be a change).
+
+### Performance, Power, and Efficiency
+
+There should not be any notable performance changes.
+
+
+## Potential Issues
+
+Outside of missing security updates and having possible dependency resolution failures due to pins, there should not be any other impacts. Considering the opt-in nature of pinning, it should not affect users unless they want to use it.
+
+## Future considerations
```

Comment:
Another consideration. As verified publishers start adding their packages, they may want to force override pins / require an update. Do we need a provision for this?

Reply:
My opinion would be that publishers shouldn't be able to override pins via the manifests, as the main reason a user would want to pin a package is because they disagree with the ISVs patching schedule. If they wanted to get around it they could add an automatic updater to their piece of software (as many have chosen to do).